### PR TITLE
[kube] support filtering for entire objects in tables

### PR DESCRIPTION
**Pull request by tigrato**
This PR resolves a bug where Teleport incorrectly returned an error when a request attempted to order resources by a field that wasn't included in the `*metav1.PartialObjectMetadata` objects returned in the table during a `kubectl get resource` operation.

Changelog: Fixes a bug in Kubernetes access that causes the error `expected *metav1.PartialObjectMetadata object` when trying to list resources.

### CODE REVIEW

**Review by fspmarshall — APPROVED** [review]

**Review by dboslee — APPROVED** [review]

File: lib/kube/proxy/resource_filters.go [hunk 1]
```diff
@@ -693,17 +693,30 @@ func (d *resourceFilterer) filterMetaV1Table(table *metav1.Table, allowedResourc
...
 			Verbs:     []string{verb},
 		}, nil
+	case namer:
+		return types.KubernetesResource{
+			Name:  o.GetName(),
+			Kind:  kind,
+			Verbs: []string{verb},
+		}, nil
 	default:
 		return types.KubernetesResource{}, trace.BadParameter("expected *metav1.PartialObjectMetadata object, got %T", obj)
```
Comment (hugoShaka) [inline lib/kube/proxy/resource_filters.go:721]: The error is not accurate anymore

File: lib/kube/proxy/resource_filters.go
Comment (hugoShaka) [inline lib/kube/proxy/resource_filters.go:1]: Do we need some test?

**Review by hugoShaka — COMMENTED** [review]
The changelog makes sense for us but it's a bit obscure for customers.

WDYT of: Fixes a bug in Kubernetes access that causes the error `Unable to list: expected *metav1.PartialObjectMetadata object` when trying to list resources.

File: lib/kube/proxy/resource_filters.go (hunk 1 above)
Comment (tigrato) [reply lib/kube/proxy/resource_filters.go]: updated

File: lib/kube/proxy/resource_filters.go
Comment (tigrato) [reply lib/kube/proxy/resource_filters.go:1]: Added in 633693773c2d479e4253116dbe7c71a011875694

**Review by hugoShaka — APPROVED** [review]

**Pull request merged**